docs/specs/multi-registry/multi_registry.md

```diff
@@ -0,0 +1,366 @@
+
+### Events
+- No event name collisions
+- Events cannot be removed
+- Events cannot "degrade" in stability (e.g., stable -> experimental)
+- The set of required/recommended attributes must remain the same
+- Event names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+### Spans
+- No span name collisions
+- Spans cannot be removed
+- Spans cannot "degrade" in stability (e.g., stable -> experimental)
+- The set of required/recommended attributes must remain the same
+- Span names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+## OTEL Semantic Convention JSON Schema Changes
+
+[TODO] The JSON Schema for the registry must be updated to reflect these changes.
+
+## OTEL Schema Changes
+
+The OTEL schema file structure must be updated to either include the URL to a self-contained/resolved
+registry or to include the resolved registry itself. 
+
+## Weaver Changes
+
+The following changes are proposed to Weaver:
+
+- Weaver must be able to support any operation on any registry (check, resolve, generate, search,
+  ...).
+- The command `weaver registry generate` must allow the generation of the referenced entities that belong to the
+  imported registries or optionally the generation of all the entities of the imported registries.
+- Extend the `--templates` parameter to allow git URL so OTEL templates (or community-based templates) can be reused
+  for any registry.
+- Extend the `--policies` parameter to allow git URL so OTEL policies (or community-based policies) can be reused for
+  any registry.
+- Maintain a list of global and local aliases and detect any conflict between them. Detect when the same schema URL
+  is imported with different aliases. Detect when an alias (global or local) is not used in the registry.
+- More TBD.
+
+Open Questions: 
+- Is a resolved registry contain any trace of the imported registries?
+- It will be nice to detect when a new registry version is available. How can we detect the presence of a new registry
+  version? A file containing of the schema URL per version for a registry? Some kind of naming conventions for the URL? 
+
+## Protocol Changes
+
+No impact on OTLP and OTAP. 
+
+A `schema_url` field is already present at the resource and scope levels. 
+
+Ideally any component of the observability pipeline should be able to fetch the resolved registry just by knowing the
+schema URL of any resource or instrumentation scope. 
+
+## OTEL SDKs Changes
+
+TBD
+
+Open Questions:
+
+- Can we enforce the presence of the schema URL in the resource and instrumentation scope?
+- How do we convey the schema URL to the SDKs? Could that be part of the codegen done by Weaver?
+
+## Resolved Semantic Convention Registry Format
+
+> Note: A resolved registry is self-contained and does not include any complex constructs
+> like `imports`, `ref`, `extends`, etc. Their **structure is less subject to change**, making them good
```

Comment:
if I'm defining a span or a metric, I have to reference some attribute and maybe extend some attribute group, so this may need clarification.